test/Microsoft.AspNetCore.Http.Tests/ResponseCookiesTest.cs

```diff
@@ -1,19 +1,24 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Xunit;
-using Microsoft.Net.Http.Headers;
+using System.Text;
 using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Net.Http.Headers;
+using Xunit;
 
 namespace Microsoft.AspNetCore.Http.Tests
 {
     public class ResponseCookiesTest
     {
+        private static readonly ObjectPool<StringBuilder> _builderPool =
+            new DefaultObjectPoolProvider().Create<StringBuilder>(new StringBuilderPooledObjectPolicy());
+
         [Fact]
         public void DeleteCookieShouldSetDefaultPath()
         {
             var headers = new HeaderDictionary();
-            var cookies = new ResponseCookies(headers);
+            var cookies = new ResponseCookies(headers, _builderPool);
```

Comment:
Some tests should use the pool and others shouldn't so you test both code paths. Or you make them all test both scenarios.